### PR TITLE
add libc building to agbcc.

### PR DESCRIPTION
**Pull request by RevoSucks**
I think travis may need to be updated to accomodate libc building?

### CODE REVIEW

File: ginclude/stddef.h [hunk 1]
```diff
@@ -18,6 +18,8 @@ typedef int wchar_t;
 
 /* A null pointer constant.  */
 
+/* in case <stdio.h> has defined it. */
```
Comment (ghost) [inline ginclude/stddef.h:21]: maybe put this comment after `#undef NULL`, on the the same line

Comment (RevoSucks) [reply ginclude/stddef.h:21]: my text editor didn't like it when I did that.

Comment (ghost) [reply ginclude/stddef.h:21]: ok. what if you use // style comments?

Comment (ghost) [reply ginclude/stddef.h:21]: or just remove the comment. putting it above both lines just makes it confusing IMO

**Comment by RevoSucks** [comment]
As we just realized in the discord, newlib (libc.a and libgcc.a) require devkitPRO r44's assembler and below in order to match Fire Red and Leaf Green. Ruby/Sapphire/Emerald is not affected by this issue.

**Pull request merged**